provisioner_test.go

```diff
@@ -119,3 +123,75 @@ func TestNewDisembedderProvisioner(t *testing.T) {
 	assert.NoError(t, err)
 	testProvisionerSuccessful(t, *p, "linux", "amd64", DefaultVersionAstilectron, DefaultVersionElectron)
 }
+
+func TestZipShouldRemove(t *testing.T) {
+	l := log.New(log.Writer(), log.Prefix(), log.Flags())
+
+	appName := "Test"
+	dataDir, _ := filepath.Abs("./testdata/tmp")
+	vendorDir := filepath.Join(dataDir, "vendor") // https://github.com/CarsonSlovoka/go-astilectron/blob/e7796e5/paths.go#L56
+	astilectronDownloadDst := filepath.Join(vendorDir, fmt.Sprintf("astilectron-v%s.zip", DefaultVersionAstilectron))
+	astilectronDir := filepath.Join(vendorDir, fmt.Sprintf("astilectron"))
+	electronDownloadDst := filepath.Join(vendorDir, fmt.Sprintf("electron-%s-%s-v%s.zip", runtime.GOOS, runtime.GOARCH, DefaultVersionElectron))
+	electronDir := filepath.Join(vendorDir, fmt.Sprintf("electron-%s-%s", runtime.GOOS, runtime.GOARCH))
+
+	t.Log("Make sure the test directory doesn't exist.")
+	if err := os.RemoveAll(dataDir); err != nil && !os.IsNotExist(err) {
+		t.FailNow()
+	}
+
+	t.Logf("Create a directory for test only:\n%s\n", dataDir)
+	if err := os.MkdirAll(dataDir, os.FileMode(0666)); err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	defer func() {
+		t.Log("Remove the test directory after the test was done.")
+		if err := os.RemoveAll(dataDir); err != nil {
+			t.Fatalf(err.Error())
+		}
+	}()
+
+	{
+		t.Log("common process")
+		a, err := New(l, Options{
+			AppName:           appName,
+			DataDirectoryPath: dataDir,
+		})
+		if err != nil {
+			t.Fatalf("main: creating astilectron failed: %s", err.Error())
+		}
+		defer a.Close()
+
+		a.HandleSignals()
+
+		if err = a.Start(); err != nil {
+			t.Fatalf("main: starting astilectron failed: %s", err.Error())
+		}
+
+		// Stop the app immediately since we care about the file only to avoid access being denied. (delete)
```

Comment:
What you can do instead of that is:

```go
go func() {
  time.Sleep(time.Second)
  a.Stop()
}()

a.Wait()
```

This will unblock once `a` has been stopped

Reply:
Thanks. I found that if I test the download file, it seems to call [a.provision] is enough. There is no need for [a.Strat()]

[a.provision]: https://github.com/asticode/go-astilectron/pull/358/commits/ae08b36c5a8f4b2b68add1fd5808ad5e8201c78a#diff-f5c6513a93e0d3a0d10fb30654a66c3ab4b7f05ed5b4dd9dd10ef379a14496adR146
[a.Strat()]: https://github.com/asticode/go-astilectron/pull/358/commits/ae08b36c5a8f4b2b68add1fd5808ad5e8201c78a#diff-f5c6513a93e0d3a0d10fb30654a66c3ab4b7f05ed5b4dd9dd10ef379a14496adL168